zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -577,32 +365,30 @@ mod tests {
 
         let expected_views = vec![
             // v_sapling_shard_unscanned_ranges
-            format!(
-                "CREATE VIEW v_sapling_shard_unscanned_ranges AS
-                SELECT
-                    shard.shard_index,
-                    shard.shard_index << 16 AS start_position,
-                    (shard.shard_index + 1) << 16 AS end_position_exclusive,
-                    IFNULL(prev_shard.subtree_end_height, {}) AS subtree_start_height,
-                    shard.subtree_end_height AS subtree_end_height,
-                    shard.contains_marked,
-                    scan_queue.block_range_start,
-                    scan_queue.block_range_end,
-                    scan_queue.priority
-                FROM sapling_tree_shards shard
-                LEFT OUTER JOIN sapling_tree_shards prev_shard
-                    ON shard.shard_index = prev_shard.shard_index + 1
-                INNER JOIN scan_queue ON 
-                    (scan_queue.block_range_start BETWEEN subtree_start_height AND shard.subtree_end_height) OR
-                    ((scan_queue.block_range_end - 1) BETWEEN subtree_start_height AND shard.subtree_end_height) OR
-                    (
-                        scan_queue.block_range_start <= prev_shard.subtree_end_height
-                        AND (scan_queue.block_range_end - 1) >= shard.subtree_end_height
-                    )
-                WHERE scan_queue.priority > {}",
-                u32::from(db_data.params.activation_height(NetworkUpgrade::Sapling).unwrap()),
-                priority_code(&ScanPriority::Scanned),
-            ),
+            "CREATE VIEW v_sapling_shard_unscanned_ranges AS
+            WITH wallet_birthday AS (SELECT MIN(birthday_height) AS height FROM accounts)
+            SELECT
+                shard.shard_index,
+                shard.shard_index << 16 AS start_position,
+                (shard.shard_index + 1) << 16 AS end_position_exclusive,
+                IFNULL(prev_shard.subtree_end_height, wallet_birthday.height) AS subtree_start_height,
```

Comment:
This should still be Sapling activation, because that is definitionally correct for the shard at index 0, whereas this change would cause the shard at index 0 to have a start height after its end height for almost all accounts. `wallet_birthday.height` should be filtering somewhere on `scan_queue.block_range_end` (or whichever is the correct bound).

Reply:
Additionally, I think we should add continuity checks for places that cause the `sapling_tree_shards` table to be modified. If we end up in a situation where the table has discontinuities in `shard_index`, this line will create rows that have incorrect and overlapping shard tree ranges, which may cause subsequent logic (that assumes correct ranges) to fail in weird and annoying ways. Functionally, this means that anyone calling `update_subtree_roots` must do so in a way that doesn't add discontinuities (as adding a root causes the entire shard to be serialized if it didn't exist).

Reply:
This hasn't been done yet, but since that's independent I think that it should be done in a separate ticket.

Reply:
Added #924

Reply:
Added #924